justfile

```diff
@@ -9,93 +9,63 @@ export CGO_ENABLED := "0"
-  just modsync
+# run st bernard directly
+stbernard *ARGS:
+  @go run github.com/specterops/bloodhound/packages/go/stbernard {{ARGS}}
 
-  echo "Ensure containers have been rebuilt"
-  if [[ "{{wipe}}" != "clean" ]]; then
-    just bh-dev build
-  else
-    echo "Clear volumes and rebuild without cache"
-    just bh-clear-volumes
-    just bh-clean-docker-build
-  fi
+# ensure dependencies are up to date
+ensure-deps *FLAGS:
+  @just stbernard deps {{FLAGS}}
 
-  echo "Start integration testing services"
-  if [[ "{{wipe}}" == "clean" ]]; then
-    echo "Clear volumes and restart testing services without cache"
-    just bh-testing-clear-volumes
-    just bh-testing build --no-cache
-  fi
+# sync modules in workspace
+modsync *FLAGS:
+  @just stbernard modsync {{FLAGS}}
 
-  echo "BloodHound CE Init Complete"
+# run code generation
+generate *FLAGS:
+  @just stbernard generate {{FLAGS}}
 
-# Show available targets for this context.
+# Show repository status
 show *FLAGS:
-  #!/usr/bin/env bash
-  set -euo pipefail
-  python3 packages/python/beagle/main.py show {{FLAGS}}
+  @just stbernard show {{FLAGS}}
 
-# Build a target. To list targets run 'just show'
-build *FLAGS:
-  #!/usr/bin/env bash
-  set -euo pipefail
-  python3 packages/python/beagle/main.py build {{FLAGS}}
+# Run all analyzers (requires jq to be installed locally)
+analyze:
+  just stbernard analysis | jq 'sort_by(.severity) | .[] | {"severity": .severity, "description": .description, "location": "\(.location.path):\(.location.lines.begin)"}'
 
-# Test a target. To list targets run 'just show'
+# Run tests
 test *FLAGS:
-  #!/usr/bin/env bash
-  set -euo pipefail
-  python3 packages/python/beagle/main.py test {{FLAGS}}
+  @just stbernard test {{FLAGS}}
 
-# sync modules in workspace
-modsync:
-  @go run github.com/specterops/bloodhound/packages/go/stbernard modsync
-
-# updates favicon.ico, logo192.png and logo512.png from logo.svg
-update-favicon:
-  @just imagemagick convert -background none ./cmd/ui/public/logo-light.svg -define icon:auto-resize ./cmd/ui/public/favicon-light.ico
-  @just imagemagick convert -background none -size 192x192 cmd/ui/public/logo-light.svg cmd/ui/public/logo-light192.png
-  @just imagemagick convert -background none -size 512x512 cmd/ui/public/logo-light.svg cmd/ui/public/logo-light512.png
-  @just imagemagick convert -background none ./cmd/ui/public/logo-dark.svg -define icon:auto-resize ./cmd/ui/public/favicon-dark.ico
-  @just imagemagick convert -background none -size 192x192 cmd/ui/public/logo-dark.svg cmd/ui/public/logo-dark192.png
-  @just imagemagick convert -background none -size 512x512 cmd/ui/public/logo-dark.svg cmd/ui/public/logo-dark512.png
+# Build application
+build *FLAGS:
+  @just stbernard build {{FLAGS}}
+
+# prepare for code review (requires jq)
+prepare-for-codereview:
+  @rm -r tmp
+  @mkdir -p tmp
+  -@just _prep-steps
+  @ echo "For more details, see output files in {{absolute_path('./tmp')}}"
+
+_prep-steps:
```

Comment:
It would be useful to add a `yarn lint` `yarn check-types` and `yarn format` to these prep steps but I am not sure if that is something you feel is in scope for this.

Reply:
So the linter is part of code analysis, and the formatter is invoked during schemagen (part of `yarn generate`). Check types is never invoked directly, but I'm pretty sure it's implicitly handled by `yarn build`. If there's a reason to run `check-types` even if the `build` is successful, I'm happy to add that step, I just don't know if the extra time is going to give us any extra visibility

Reply:
Ah, that makes sense and you're right the type check would be handled by build. Nvm, looks like you got it covered :D

Reply:
Awesome, glad my assumptions were correct. Thanks for validating, gives me a lot more confidence in this workflow